kura/org.eclipse.kura.core.comm/src/main/java/org/eclipse/kura/core/comm/CommConnectionFactory.java

```diff
@@ -8,49 +8,26 @@
  *
  * Contributors:
  *     Eurotech
+ *     Red Hat Inc - clean up
  *******************************************************************************/
 package org.eclipse.kura.core.comm;
 
+import static org.eclipse.kura.comm.CommURI.parseString;
+
 import java.io.IOException;
 
 import javax.microedition.io.Connection;
 
-import org.eclipse.kura.comm.CommURI;
-import org.osgi.service.component.ComponentContext;
 import org.osgi.service.io.ConnectionFactory;
 
 public class CommConnectionFactory implements ConnectionFactory {
 
-    @SuppressWarnings("unused")
-    private ComponentContext m_ctx;
-
-    // ----------------------------------------------------------------
-    //
-    // Dependencies
-    //
-    // ----------------------------------------------------------------
-
-    // ----------------------------------------------------------------
-    //
-    // Activation APIs
-    //
-    // ----------------------------------------------------------------
-
-    protected void activate(ComponentContext componentContext) {
-        //
-        // save the bundle context
-        this.m_ctx = componentContext;
-    }
-
-    protected void deactivate(ComponentContext componentContext) {
-        this.m_ctx = null;
-    }
-
     @Override
     public Connection createConnection(String name, int mode, boolean timeouts) throws IOException {
         try {
-            CommURI uri = CommURI.parseString(name);
-            return new CommConnectionImpl(uri, mode, timeouts);
+            return new CommConnectionImpl(parseString(name), mode, timeouts);
+        } catch (IOException e) {
+            throw e; // re-throw
         } catch (Throwable t) {
```

Comment:
Actually the usual way to throw a checked exception in kura is by wrapping it in KuraException. And in case of unchecked one, the usual way is to wrap in KuraRuntimeException.